chromatic_fitting/models/eclipse.py

```diff
@@ -0,0 +1,505 @@
+            print(
+                "Starry will not play nice with pymc3 in greedy mode.\n Please set starry.config.lazy=True and try again!"
+            )
+            return
+
+        # ensure that attach data has been run before setup_lightcurves
+        if not hasattr(self, "data"):
+            print("You need to attach some data to this chromatic model!")
+            return
+
+        # if the model has a name then add this to each parameter's name
+        if hasattr(self, "name"):
+            name = self.name + "_"
+        else:
+            name = ""
+            print("No name set for the model.")
+
+        # set up the models, data and orbits in a format for looping
+        if self.optimization == "separate":
+            models = self._pymc3_model
+            datas = [self.get_data(i) for i in range(self.data.nwave)]
+        else:
+            models = [self._pymc3_model]
+            datas = [self.get_data()]
+
+        kw = {"shape": datas[0].nwave}
+
+        # if the .every_light_curve attribute (final lc model) is not already present then create it now
+        if not hasattr(self, "every_light_curve"):
+            self.every_light_curve = {}
+
+        # we can decide to store the LC models during the fit (useful for plotting later, however, uses large amounts
+        # of RAM)
+        if store_models == True:
+            self.store_models = store_models
+
+        parameters_to_loop_over = {
+            f"{name}stellar_amplitude": [],
+            f"{name}stellar_radius": [],
+            f"{name}stellar_mass": [],
+            f"{name}stellar_prot": [],
+            f"{name}period": [],
+            f"{name}t0": [],
+            f"{name}planet_log_amplitude": [],
+            f"{name}inclination": [],
+            f"{name}planet_mass": [],
+            f"{name}planet_radius": [],
+            f"{name}eccentricity": [],
+            f"{name}omega": [],
+            f"{name}limbdark1": [],
+            f"{name}limbdark2": [],
+        }
+
+        self.initial_flux_model_guess = []
+
+        for j, (mod, data) in enumerate(zip(models, datas)):
+            if self.optimization == "separate":
+                kw["i"] = j
+
+            with mod:
+                for pname in parameters_to_loop_over.keys():
+                    parameters_to_loop_over[pname].append(
+                        self.parameters[pname].get_prior_vector(**kw)
+                    )
+
+                y_model = []
+
+                for i, w in enumerate(data.wavelength):
+                    param_i = {}
+                    for param_name, param in parameters_to_loop_over.items():
+                        #                     if isinstance(self.parameters[param_name], WavelikeFitted):
+                        try:
+                            param_i[param_name] = param[j][i]
+                        except:
+                            try:
+                                param_i[param_name] = param[j][0]
```

Comment:
had to add this line - otherwise the simultaneous fitting did not work. Test out in other models...